task_templates/pipelines/python3_calibrated_anomaly_detection/custom.py

```diff
@@ -1,40 +1,111 @@
+"""
+    In this example we show how to create an unsupervised anomaly detection model
+    with calibrated predictions
+
+    Note: in our score hook we use the same column name as regression, i.e. columns=['Predictions']
+"""
+from typing import List, Optional, Any, Dict
+import pandas as pd
+import numpy as np
 import pickle
 
 from build_pipeline import make_anomaly
 
 
 def fit(
-    X, output_dir, class_order=None, row_weights=None, **kwargs,
-):
-    """
-    This hook must be implemented with your fitting code, for running drum in the fit mode.
-    This hook MUST ALWAYS be implemented for custom tasks.
-    For inference models, this hook can stick around unimplemented, and won’t be triggered.
+        X: pd.DataFrame,
+        y: pd.Series,
+        output_dir: str,
+        class_order: Optional[List[str]] = None,
+        row_weights: Optional[np.ndarray] = None,
+        **kwargs,
+) -> None:
+    """ This hook MUST ALWAYS be implemented for custom tasks.
+
+    This hook defines how DataRobot will train this task.
+    DataRobot runs this hook when the task is being trained inside a blueprint.
+
+    DataRobot will pass the training data, project target, and additional parameters based on the project
+    and blueprint configuration as parameters to this function.
+
+    As an output, this hook is expected to create an artifact containing a trained object,
+    that is then used to score new data.
+
     Parameters
     ----------
-    X: pd.DataFrame - training data to perform fit on
-    y: pd.Series - target data to perform fit on
-    output_dir: the path to write output. This is the path provided in '--output' parameter of the
-        'drum fit' command.
-    class_order : A two element long list dictating the order of classes which should be used for
-        modeling. This will not be used for anomaly detection models
-    row_weights: An array of non-negative numeric values which can be used to dictate how important
+    X: pd.DataFrame
+        Training data that DataRobot passes when this task is being trained. Note that both the training data AND
+        column (feature) names are passed
+    y: pd.Series
+        Project's target column.
```

Comment:
here, for the anomaly pipeline, we should probably add that in case of anomaly detection, None is passed